lib/puppet/provider/rabbitmq_binding/rabbitmqadmin.rb

```diff
@@ -37,10 +37,10 @@ def self.instances
         if arguments.nil?
           arguments = '{}'
         else
-          arguments = arguments.gsub(%r{^\[(.*)\]$}, '').gsub(%r{\{("(?:.|\\")*?"),}, '{\1:').gsub(%r{\},\{}, ',')
+          arguments = arguments.gsub(%r{^\[|\]$}, '').gsub(%r{\{("(?:.|\\")*?"),}, '{\1:').gsub(%r{\},\{}, ',')
```

Comment:
Might be good to add a comment here explaining the logic, since this is pretty obtuse / hard to read... Where is `\1` in `{\1:` coming from now that the capturing parenthesis are gone? is it still needed?

Reply:
The second substitution has `?:`, so should be non-capturing, right? What is getting pulled into `\1` exactly, especially if it used to be coming from the first capturing parens that are now gone?